server/src/main/java/org/elasticsearch/common/component/AbstractComponent.java

```diff
@@ -26,10 +26,8 @@
 public abstract class AbstractComponent {
 
     protected final Logger logger;
-    protected final Settings settings;
 
     public AbstractComponent(Settings settings) {
```

Comment:
I'm intentionally keeping the `settings` argument because removing it is bigger than this change. We do much more plumbing of this object then we do using it.